etc/di.xml

```diff
@@ -112,4 +112,10 @@
       <argument name="filesystem" xsi:type="object">Magento\Framework\Filesystem\Driver\File</argument>
     </arguments>
   </type>
+
+  <type name="Paytrail\PaymentService\Model\Company\CompanyCommerceRequestData">
+    <arguments>
+      <argument name="object" xsi:type="object">Magento\Company\Model\CompanyRepository</argument>
```

Comment:
Injecting a class that does not exist in every scenario into a di.xml file can have unintended consequences. I applaud the creativity of using a proxy to prevent instantiation of this class in open source editions. But I do not know enough about the di.xml system and every scenario it runs through to be able to approve this approach into a generic module. 

My main questions are:
- Is there a flow that can trigger the parsing and object look-up of this declaration without the class being loaded. If it happens the result could potentially be a compilation or runtime error that cannot be recovered from.
- Why does the declaration require the name object (referring to a slack conversation here) What inbuilt system are we abusing to make it work? And how reliable is it?

Personally I would like to refactor this approach but we can perhaps take a look at that together. After discussing with project manager.